src/verifier.rs

```diff
@@ -305,6 +333,38 @@ impl SDJWTVerifier {
 
         Ok(())
     }
+
+    fn unpack_from_digest(
+        &mut self,
+        digest: &Value,
+    ) -> Result<Option<Value>> {
+        let digest = digest
+            .as_str()
+            .ok_or(Error::ConversionError("str".to_string()))?;
+        if self.duplicate_hash_check.contains(&digest.to_string()) {
+            return Err(Error::DuplicateDigestError(digest.to_string()));
+        }
+        self.duplicate_hash_check.push(digest.to_string());
+
+        if let Some(value_for_digest) =
+            self.sd_jwt_engine.hash_to_decoded_disclosure.get(digest)
+        {
+            let disclosure =
+                value_for_digest
+                    .as_array()
+                    .ok_or(Error::InvalidArrayDisclosureObject(
+                        value_for_digest.to_string(),
+                    ))?;
+
+            let value = disclosure[1].clone();
```

Comment:
Let's check here the length of the disclosure array. And let's add similar change in `upack_from_digests`